src/llmcompressor/modifiers/awq/base.py

```diff
@@ -220,64 +324,47 @@ def _set_resolved_mappings(self, model: Module) -> None:
         self._resolved_mappings = resolved_mappings
         return
 
-    def _setup_scale_hooks(self) -> None:
+    def _setup_activation_cache_hooks(self) -> None:
         """
         Attach a forward hook to each activation we want to smooth. This allows us to
         calculate the dynamic range during calibration
         """
 
-        def create_hook_fn(layer_name):
-            def hook_fn(module, inp, out):
-                inp = inp[0].cpu().detach()
-
-                if layer_name in self._scales:
-                    self._scales[layer_name].append(inp)
+        def create_cache_activation_hook(layer_name):
+            def cache_activation_hook_fn(
```

Comment:
nit: You can just define this hook as a class method, rather than creating a new instance of the function every time

Reply:
Is there a performance hit to doing that? If not, I prefer this as more scoped

Reply:
Also note the `layer_name` input, i think we'll have to create a function every time because of this

Reply:
Ah yeah, in other implementations I curry this using `partial` but this is good too

Reply:
🍛